File: Assets/Scripts/Minesweeper/Core/IMinesweeperService.cs
using System;

namespace Kukumberman.Minesweeper.Core
{
    public interface IMinesweeperService
    {
        event Action OnStateChanged;

        ref MinesweeperCell CellAt(int index);

        int CellCount { get; }

        int BombCount { get; }

        int Width { get; }

        int Height { get; }

        EMinesweeperState State { get; }

        void Dispose();

        void StartGame(MinesweeperGameSettings settings, int seed);

        void Restart();

        void RevealCell(int index);

        void FlagCell(int index);
    }
}

File: react/src/core/minesweeper/IMinesweeperService.ts
import MinesweeperGame from "./MinesweeperGame";
import { MinesweeperState } from "./enums";

export default interface IMinesweeperService {
  Game: MinesweeperGame;
  State: MinesweeperState;
  StartGame(settings: MinesweeperGameSettings, seed: number): void;
  Restart(): void;
  RevealCell(index: number): void;
  FlagCell(index: number): void;
}
